code/game/machinery/vending/cm_vending.dm

```diff
@@ -95,6 +97,71 @@ IN_USE						used for vending/denying
 			return
 	return
 
+// Vendor Icon Procs
+
+/obj/structure/machinery/cm_vending/proc/preload_assets()
+	var/datum/asset/simple/dynamic_icons/dyn = get_asset_datum(/datum/asset/simple/dynamic_icons)
+	for (var/list/i in product_icon_list)
```

Comment:
I believe you can use `as anything` here since it's always going to be lists in the list

Reply:
This seems to cause an issue